analyses/snv-callers/lancet-wxs-tests/explore-tcga-pbta.Rmd

```diff
@@ -0,0 +1,591 @@
+
+  # Make filename to save plot as
+  plot.file <- file.path(plots_dir, paste0(plot_name, ".png"))
+
+  # Make the Venn diagram
+  grid::grid.newpage()
+  venn.plot <- VennDiagram::draw.pairwise.venn(
+    area1 = sum(bed_granges_1@ranges@width),
+    area2 = sum(bed_granges_2@ranges@width),
+    cross.area = sum(overlaps@ranges@width),
+    category = c(name_1, name_2),
+    fill = c("#F8766D", "#00BFC4"),
+    cex = 2,
+    cat.cex = 1.5,
+    ext.pos = 0,
+    ext.dist = -0.01,
+    ext.length = .8,
+    ext.line.lwd = 2,
+    ext.line.lty = "dashed",
+    margin = 0.1
+  )
+  grid::grid.draw(venn.plot)
+
+  # Save as a PNG
+  png(plot.file)
+  grid::grid.draw(venn.plot)
+  dev.off()
+
+  # Print out a summary of the ratios
+  cat(
+    " Ratio of", name_1, "overlapped:",
+    sum(overlaps@ranges@width) / sum(bed_granges_1@ranges@width), "\n",
+    "Ratio of", name_2, "overlapped:",
+    sum(overlaps@ranges@width) / sum(bed_granges_2@ranges@width), "\n"
+  )
+}
+```
+
+## Read in the metadata. 
+
+```{r}
+pbta_metadata <- readr::read_tsv(file.path(data_dir, "pbta-histologies.tsv"))
+tcga_metadata <- readr::read_tsv(file.path(data_dir, "pbta-tcga-manifest.tsv")) %>%
+  dplyr::mutate(
+    experimental_strategy = "WXS", # This field doesn't exist for this data, but all is WXS
+    short_histology = Primary_diagnosis
+  ) # This field is named differently in PBTA
+```
+
+## Set up the datasets from the databases
+
+Declare the columns we'll keep. 
+
+```{r}
+cols_to_keep <- c(
+  "Chromosome",
+  "Start_Position",
+  "End_Position",
+  "Reference_Allele",
+  "Allele",
+  "Tumor_Sample_Barcode",
+  "Variant_Classification",
+  "t_depth",
+  "t_ref_count",
+  "t_alt_count",
+  "n_depth",
+  "n_ref_count",
+  "n_alt_count",
+  "VAF"
+)
+```
+
+Set up TCGA data. 
+
+```{r}
+tcga_lancet <- set_up_tcga("lancet")
+tcga_mutect <- set_up_tcga("mutect")
+tcga_strelka <- set_up_tcga("strelka")
+tcga_consensus <- set_up_tcga("consensus")
+```
+
+Set up PBTA data. 
+
+```{r}
+pbta_lancet <- set_up_pbta("lancet")
+pbta_mutect <- set_up_pbta("mutect")
+pbta_strelka <- set_up_pbta("strelka")
+pbta_consensus <- set_up_pbta("consensus")
+```
+
+## How does the tumor sequencing depth compare for each TCGA and PBTA data?
+
+Set up combined PBTA and TCGA data.frames. 
+
+```{r}
+lancet <- dplyr::bind_rows(list(
+  tcga = tcga_lancet,
+  pbta = pbta_lancet
+), .id = "dataset") %>%
+  dplyr::mutate(dataset = as.factor(dataset))
+
+mutect <- dplyr::bind_rows(list(
+  tcga = tcga_mutect,
+  pbta = pbta_mutect
+), .id = "dataset") %>%
+  dplyr::mutate(dataset = as.factor(dataset))
+
+strelka <- dplyr::bind_rows(list(
+  tcga = tcga_strelka,
+  pbta = pbta_strelka
+), .id = "dataset") %>%
+  dplyr::mutate(dataset = as.factor(dataset))
+```
+
+We'll plot the tumor sequencing depth for each caller as a series of density plots. 
+
+```{r}
+lancet %>%
+  ggplot2::ggplot(ggplot2::aes(x = log10(t_depth), color = dataset)) +
+  ggplot2::geom_density()
```

Comment:
The bandwidth seems too small for these plots, and inconsistent, using `geom_density(bw = 1)` might be a good idea, though you will want to play with bandwidth to get a reasonable result.